### PR TITLE
Resolve conflicts in asmcomp/*cmm* and middle_end/

### PR DESCRIPTION
**Pull request by mshinwell**
Still to do (at least):
- check compilation
- check other places we might need to change in `Cmm_helpers` e.g. to use `get_field_codepointer` or `mk_load_mut`
- fix `Compilenv` around the place the new check for packing was added, this isn't correct yet

Includes #171

### CODE REVIEW

**Comment by ncik-roberts** [comment]
Let's rebase this after #172 is merged. That's just the mli pieces.

File: middle_end/printclambda_primitives.ml [hunk 1]
```diff
@@ -117,27 +108,14 @@ let primitive ppf (prim:Clambda_primitives.primitive) =
...
-      fprintf ppf "makeblock %i%a" tag Printlambda.block_shape shape
-  | Pmakeblock(tag, Mutable, shape) ->
-      fprintf ppf "makemutable %i%a" tag Printlambda.block_shape shape
-  | Pfield(n, ptr, mut) ->
+  | Pfield (n, layout, ptr, mut) ->
       let instr =
         match ptr, mut with
         | Immediate, _ -> "field_int "
         | Pointer, Mutable -> "field_mut "
         | Pointer, Immutable -> "field_imm "
```
Comment (Ekdohibs) [inline middle_end/printclambda_primitives.ml:116]: We should probably remove the trailing spaces in the strings there.

**Review by Ekdohibs — COMMENTED** [review]
Mostly good, see comments.

**Comment by mshinwell** [comment]
This is now done and ready for final review.  cc @lukemaurer for `Compilenv` changes

File: asmcomp/cmmgen.ml [hunk 2]
```diff
@@ -978,8 +978,15 @@ and transl_prim_1 env p arg dbg =
   | Pget_header m ->
       box_int dbg Pnativeint m (get_header (transl env arg) dbg)
   | Pperform ->
-      let cont = make_alloc dbg Obj.cont_tag [int_const dbg 0] in
-      Cop(Capply typ_val,
+      (* We have to hard-code this at the moment since the system compiler's
+         stdlib doesn't have [Obj.cont_tag]. *)
+      let cont_tag = 245 in
+      let cont =
```
Comment (Ekdohibs) [inline asmcomp/cmmgen.ml:984]: Please put a fatal error here as well; by the time this becomes relevant, we should at least have access to `runtimetags.ml` (https://github.com/ocaml-flambda/flambda-backend/pull/1857) which would make this safer.

Comment (mshinwell) [reply asmcomp/cmmgen.ml]: Fixed, and added fatal errors in the other cases

**Review by Ekdohibs — APPROVED** [review]

**Pull request merged**